hybrid/decomposers.py

```diff
@@ -44,6 +45,91 @@ class IdentityDecomposer(traits.ProblemDecomposer, traits.SISO, Runnable):
     def next(self, state, **runopts):
         return state.updated(subproblem=state.problem)
 
+class ComponentDecomposer(traits.ProblemDecomposer, traits.SISO, Runnable):
+    """Selects a subproblem of variables that make up a connected component.
+
+    Args:
+        rolling (bool, optional, default=True):
+            If True, successive calls for the same problem (with possibly
+            different samples) produce subproblems on different components,
+            selected by rolling down the list of all components sorted by
+            the `key` function.
+
+        key (function, optional, default=len):
```

Comment:
[`callable`](https://docs.python.org/3/library/functions.html#callable) is more [general](https://docs.python.org/3/reference/datamodel.html#object.__call__) than a function. A we actually accept a callable here.